File: src/main/java/com/yoghurt1131/weatherapi/application/controller/CurrentWeatherController.java
package com.yoghurt1131.weatherapi.application.controller;

import com.yoghurt1131.weatherapi.application.Exception.ApiCallException;
import com.yoghurt1131.weatherapi.application.service.WeatherApiService;
import com.yoghurt1131.weatherapi.domain.CurrentWeather;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RequestMapping("/current")
@RestController
public class CurrentWeatherController {

    @Autowired
    WeatherApiService weatherApiService;

    @GetMapping("/city")
    public CurrentWeather city(@RequestParam(required = true) String cityName) {
        try {
            CurrentWeather weather = weatherApiService.getCurrentWeather(cityName);
            return weather;
        } catch (ApiCallException e) {
            return null;
        }
    }
}

File: src/main/kotlin/dev/yoghurt1131/weatherapi/application/controller/CurrentWeatherController.kt
package dev.yoghurt1131.weatherapi.application.controller

import dev.yoghurt1131.weatherapi.application.service.WeatherApiService
import dev.yoghurt1131.weatherapi.application.controller.response.CurrentWeather
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RequestMapping("/current")
@RestController
class CurrentWeatherController(private val weatherApiService: WeatherApiService) {

    private val logger = LoggerFactory.getLogger(this.javaClass)

    @GetMapping("/city")
    fun city(cityName: String): CurrentWeather = weatherApiService.getCurrentWeather(cityName)
}
